conjur/cli.py

```diff
@@ -706,33 +707,37 @@ def run_action(resource, args):
 
         credential_provider, _ = CredentialStoreFactory.create_credential_store()
         # pylint: disable=no-else-return,line-too-long
+        if resource == 'logout':
+            Cli.handle_logout_logic(credential_provider, args.ssl_verify)
+            sys.stdout.write("Successfully logged out from Conjur\n")
+            return
+
         if resource == 'init':
             Cli.handle_init_logic(args.url, args.name, args.certificate, args.force, args.ssl_verify)
             # A successful exit is required to prevent the initialization of
             # the Client because the init command does not require the Client
             return
-        elif resource == 'login':
+
+        # Needed for unit tests so that they do not require configuring
+        is_testing_env = os.getenv('TEST_ENV') in ('true', 'True')
+
+        # If the user runs a command without configuring the CLI,
+        # we request they do so before executing their request
+        # pylint: disable=line-too-long
+        if not is_testing_env and file_is_missing_or_empty(DEFAULT_CONFIG_FILE):
+            sys.stdout.write("The Conjur CLI needs to be initialized before you can use it.\n")
+            Cli.handle_init_logic()
+
+        if resource == 'login':
             Cli.handle_login_logic(credential_provider, args.identifier, args.password, args.ssl_verify)
             return
-        elif resource == 'logout':
-            Cli.handle_logout_logic(credential_provider, args.ssl_verify)
-            sys.stdout.write("Successfully logged out from Conjur\n")
-            return
 
-        # Needed for unit tests so that they do not require configuring
-        if os.getenv('TEST_ENV') is None or os.getenv('TEST_ENV') == 'False':
-            # If the user runs a command without configuring the CLI,
-            # we request they do so before executing their request
-            # pylint: disable=line-too-long
-            if not os.path.exists(DEFAULT_CONFIG_FILE) or os.path.getsize(DEFAULT_CONFIG_FILE) == 0:
-                sys.stdout.write("Error: The Conjur CLI has not been initialized\n")
-                Cli.handle_init_logic()
-
-            # If the user runs a command without logging into the CLI,
-            # we request they do so before executing their request
+        # If the user runs a command without logging into the CLI,
+        # we request they do so before executing their request
+        if not is_testing_env:
             loaded_conjurrc = ConjurrcData.load_from_file()
             if not credential_provider.is_exists(loaded_conjurrc.conjur_url):
-                sys.stdout.write("Error: You have not logged in\n")
+                sys.stdout.write("%s\n" % LOGIN_IS_REQUIRED)
```

Comment:
%s or fstring?

Reply:
do you think we should do this for every user interaction string?

Reply:
sys.stdout.write(f"{LOGIN_IS_REQUIRED}\n")

pylint will probably fail so add `# pylint: disable=logging-fstring-interpolation` above